src/simulations/results/results_plotting_utils.jl

```diff
@@ -0,0 +1,406 @@
+         else
+             fig = Figure(layout=GridLayout(1, 3)) 
+             ax = Axis(fig[1, 3], xlabel="Δ$variable_title ($(title_mapping[string(variables[1])][2]))", ylabel="Frequency", title="Histogram of $(title_mapping[string(variables[1])][1]) Evolution")
+             ax_diff = Axis(fig[1, 1], title="$(title_mapping[string(variables[1])][1]) Evolution",width=diff_width,height=diff_height)
+         end
+    
+        # Plot based on the metric
+        for metric in metrics
+            if metric == "hist"
+                hist!(ax, vec(data[end]-data[1]), bins=50)
+                ax.limits[] = (minimum(data_diff), maximum(data_diff), nothing, nothing)
+            elseif metric == "difference"
+                # Mirror data around x and y
+                data_diff = reverse(reverse(data_diff, dims=1), dims=2)
+                
+                hm_diff = heatmap!(ax_diff, data_diff, colormap=:redsblues, color=:auto, halign=:right)
+                ax_diff.xlabel = "X coordinate"
+                ax_diff.ylabel = "Y coordinate"
+                Colorbar(fig[1, 2], hm_diff)
+              
+                
+            end
+        end
+    
+        fig  # Return the main figure
+    
+end
+
+"""
+    plot_glacier_statistics(
+        results::T, 
+        variables::Vector{Symbol}; 
+        tspan::Tuple{Float64, Float64}, 
+        metrics::Vector{String} = ["average"], 
+        threshold_ratio::Float64 = 0.001
+    ) -> Figure
+
+Plot statistical metrics for a specific glacier variable over a given time span.
+
+# Arguments
+- `results`: A custom type containing the results of a glacier simulation.
+  * Must have fields corresponding to the variables to be plotted.
+- `variables`: A vector of symbols representing the variable to be plotted. Only one variable is allowed.
+  * Possible variables: `:H`, or any other data that has time evolution in it.
+- `tspan`: A tuple representing the start and end time for the simulation.
+- `metrics`: A vector of strings indicating the statistical metrics to visualize. 
+  * Possible values: "average", "median", "min", "max", "std". Defaults to `["average"]`.
+- `threshold_ratio`: A float specifying the threshold ratio for filtering data. Data below this threshold is ignored in the statistical calculations.
+
+# Returns
+- A `Figure` object containing the statistical metrics for the specified glacier variable.
+
+# Notes
+- The function uses predefined titles for each variable based on a mapping, any other variable just get their symbol as title.
+- Only one variable can be plotted at a time with this function.
+- The threshold for filtering is calculated based on the maximum value in the data and the specified threshold_ratio.
+"""
+function plot_glacier_statistics(results::T, variables::Vector{Symbol}; tspan::Tuple{Float64, Float64}, metrics::Vector{String} = ["average"], threshold_ratio::Float64 = 0.001)::Figure where T
```

Comment:
I think this plot should be called `plot_glacier_evolution_stats`, just to convey the fact that you are computing statistics on how the glacier evolved through time.

Reply:
Changed to convenient name